### PR TITLE
feat(agent): Add SSH max timeout option for coder agent

### PR DESCRIPTION
**Pull request by bensejas**
## Context
We want to users re-authenticate after a set period of time. This is achievable on the browser side but not once users are connected through SSH.

## Changes
Add an option to set a max timeout for SSH connections on the coder agent. 

## Future work
* Update terraform provider so max timeout option is configurable https://github.com/coder/terraform-provider-coder/blob/main/provider/agent.go
* Update bootstrap scripts with a ssh max timeout environment variable `./provisioner/scripts/*`

### CODE REVIEW

**Comment by bensejas** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by kylecarbs** [comment]
Exciting! It'd be nice to print out to TTY sessions why the connection is closing as well.

Would adding an additional authentication step to keep a connection alive be of interest? We could make this experience super slick in the VS Code extension as well... essentially just hanging all SSH connections until the user refreshes an authentication token.

**Comment by bensejas** [comment]
>It'd be nice to print out to TTY sessions why the connection is closing as well.
I'll have a look 👀 

> Would adding an additional authentication step to keep a connection alive be of interest?
That would be nice user experience. We are using a mix of VS Code and Intellij at the moment but anything would be nice.

**Comment by bensejas** [comment]
Awesome, that would be good. Thanks @kylecarbs

File: agent/agent.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ type Options struct {
 	EnvironmentVariables   map[string]string
 	Logger                 slog.Logger
 	AgentPorts             map[int]string
+	SshMaxTimeout          time.Duration
```
Comment (kylecarbs) [inline agent/agent.go:81]:
```suggestion
	SSHMaxTimeout          time.Duration
```

**Review by kylecarbs — APPROVED** [review]
I was going to suggest we add a test, but I don't think that's overly required since the value is just being piped through.

Be weary that users may report disconnection issues though, because this will just terminate things!

**Comment by bensejas** [comment]
@kylecarbs I fixed up the linting and updated the test snapshot. It looks like I don't have permission to run the test workflows to check.

**Comment by kylecarbs** [comment]
Approved! I'll merge once passing.

**Pull request merged**